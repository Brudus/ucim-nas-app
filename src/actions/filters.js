// SET_TEXT_FILTER
export const setTextFilter = (text = '') => ({
    type: 'SET_TEXT_FILTER',
    text
});

// SORT_BY_SOURCE
export const sortBySource = () => ({
    type: 'SORT_BY_SOURCE'
});

// SORT_BY_DESTINATION
export const sortByDestination = () => ({
    type: 'SORT_BY_DESTINATION'
});

// SORT_BY_REPEAT_AT
export const sortByRepeatAt = () => ({
    type: 'SORT_BY_REPEAT_AT'
});