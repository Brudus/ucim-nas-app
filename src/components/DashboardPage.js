import React from 'react';

const DashboardPage = () => (
    <div>
        <p>You are logged in!</p>
    </div>
);

export default DashboardPage;