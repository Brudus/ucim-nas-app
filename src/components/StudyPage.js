import React from 'react';
import { connect } from 'react-redux';

export const StudyPage = (props) => (
    <div>
        <h2>Study!</h2>
        {props.words[0].source}
    </div>
);

const mapStateToProps = (state) => ({
    words: state.words
});

export default connect(mapStateToProps)(StudyPage);