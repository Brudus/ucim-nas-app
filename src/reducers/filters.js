const filtersReducerDefaultState = {
    text: '',
    sortBy: 'destinationText'
};

export default (state = filtersReducerDefaultState, action) => {
    switch(action.type) {
        case 'SET_TEXT_FILTER':
            return {
                ...state,
                text: action.text
            };
        case 'SORT_BY_SOURCE':
            return {
                ...state,
                sortBy: 'source'
            };
        case 'SORT_BY_DESTINATION':
            return {
                ...state,
                sortBy: 'destination'
            };
        case 'SORT_BY_REPEAT_AT':
            return {
                ...state,
                sortBy: 'repeatAt'
            };
        default:
            return state;
    }
};