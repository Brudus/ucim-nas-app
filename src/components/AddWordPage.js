import React from 'react';
import { connect } from 'react-redux';
import WordForm from './WordForm';
import { startAddWord } from '../actions/words';

export class AddWordPage extends React.Component {
    onSubmit = (word) => {
        this.props.startAddWord(word);
    };

    render() {
        return (
            <div>
                <h1>Add Word</h1>
                <WordForm 
                    onSubmit={this.onSubmit}
                />
            </div>
        );
    };
}

const mapDispatchToProps = (dispatch) => ({
    startAddWord: (word) => dispatch(startAddWord(word))
});

export default connect(undefined, mapDispatchToProps)(AddWordPage);