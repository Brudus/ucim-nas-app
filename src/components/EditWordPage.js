import React from 'react';
import { connect } from 'react-redux';
import WordForm from './WordForm';
import { startEditWord, startRemoveWord } from '../actions/words';

export class EditWordPage extends React.Component {
    onSubmit = (word) => {
        this.props.startEditWord(this.props.word.id, word);
        this.props.history.push('/catalog');
    };
    onRemove = (id) => {
        this.props.startRemoveWord({ id: this.props.word.id });
        this.props.history.push('/catalog');
    };

    render() {
        return (
            <div>
                <WordForm 
                    word={this.props.word}
                    onSubmit={this.onSubmit}
                />
                <button onClick={this.onRemove}>Remove</button>
            </div>
        );
    };
};

const mapStateToProps = (state, props) => ({
    word: state.words.find((word) => word.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditWord: (id, word) => dispatch(startEditWord(id, word)),
    startRemoveWord: (data) => dispatch(startRemoveWord(data))
});

export default connect(mapStateToProps, mapDispatchToProps)(EditWordPage);