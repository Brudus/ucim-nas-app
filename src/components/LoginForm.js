import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export class LoginForm extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            email: '',
            password: '',
            loginError: '',
            emailPassed: undefined
        };
    }

    validateEmail = (email) => {
        const regex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
        return regex.test(email);
    };

    onEmailChange = (e) => {
        const email = e.target.value;
        const emailPassed = this.validateEmail(email);
        this.setState(() => ({ 
            email,
            emailPassed
        }));
    };

    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() => ({ password }));
    }

    onSubmit = (e) => {
        e.preventDefault();

        this.setState(() => ({ 
            loginError: '',
            emailPassed: true
        }));
        this.props.startLogin(this.state.email, this.state.password);
    };

    render() {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    {this.state.loginError && <p>{this.state.loginError}</p>}
                    <div>
                        <input 
                            type="email"
                            placeholder="Email address" 
                            autoFocus
                            onChange={this.onEmailChange}
                        />
                    </div>
                    
                    <div>
                        <input 
                            type="password"
                            placeholder="Password" 
                            onChange={this.onPasswordChange}
                        />
                    </div>
                    <button disabled={!this.state.emailPassed || !this.state.password}>Login</button>
                </form> 
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch, props) => ({
    startLogin: (email, password) => dispatch(startLogin(email, password))
});

export default connect(undefined, mapDispatchToProps)(LoginForm);