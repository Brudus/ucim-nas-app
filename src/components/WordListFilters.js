import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortBySource, sortByDestination, sortByRepeatAt } from '../actions/filters';

const WordListFilters = (props) => (
    <div>
        <input type="text" value={props.filters.text} onChange={(e) => {
            props.dispatch(setTextFilter(e.target.value));
        }} />
        <select value={props.filters.sortBy} onChange={(e) => {
            switch (e.target.value) {
                case 'source':
                    props.dispatch(sortBySource());
                    break;
                case 'destination':
                    props.dispatch(sortByDestination());
                    break;
                case 'repeatAt':
                    props.dispatch(sortByRepeatAt());
                    break;
            }
        }}>
            <option value="destination">Serbo-Croatian</option>
            <option value="source">German</option>
            <option value="repeatAt">Repetition date</option>
        </select>
    </div>
);

const mapStateToProps = (state) => {
    return {
        filters: state.filters
    };
};

export default connect(mapStateToProps)(WordListFilters);