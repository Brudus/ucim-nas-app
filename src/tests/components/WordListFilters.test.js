import React from 'react';
import { shallow } from 'enzyme';
import { WordListFilters } from '../../components/WordListFilters';
import { defaultFilters, changedFilters } from '../fixtures/filters';

let setTextFilter, sortByDestination, sortBySource, sortByRepeatAt, wrapper;

beforeEach(() => {
    setTextFilter = jest.fn();
    sortByDestination = jest.fn();
    sortBySource = jest.fn();
    sortByRepeatAt = jest.fn();
    wrapper = shallow(
        <WordListFilters 
            filters={defaultFilters}
            setTextFilter={setTextFilter}
            sortBySource={sortBySource}
            sortByDestination={sortByDestination}
            sortByRepeatAt={sortByRepeatAt}
        />
    );
});

test('should render WordListFilters correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should render WordListFilters with changed filters correclty', () => {
    wrapper.setProps({
        filters: changedFilters
    });
    expect(wrapper).toMatchSnapshot();
});

test('should handle text change', () => {
    const value = 'd';
    const event = {
        target: { value }
    };
    wrapper.find('input').prop('onChange')(event);
    expect(setTextFilter).toHaveBeenLastCalledWith(value);
});

test('should sort by destination', () => {
    const value = 'destination';
    wrapper.setProps({
        filters: changedFilters
    })
    wrapper.find('select').simulate('change', {
        target: { value }
    });
    expect(sortByDestination).toHaveBeenCalled();
});

test('should sort by source', () => {
    const value = 'source';
    wrapper.find('select').simulate('change', {
        target: { value }
    });
    expect(sortBySource).toHaveBeenCalled();
});

test('should sort by repeatAt', () => {
    const value = 'repeatAt';
    wrapper.find('select').simulate('change', {
        target: { value }
    });
    expect(sortByRepeatAt).toHaveBeenCalled();
});