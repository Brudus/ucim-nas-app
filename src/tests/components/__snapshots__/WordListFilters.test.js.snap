// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render WordListFilters correctly 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <select
    onChange={[Function]}
    value="destination"
  >
    <option
      value="destination"
    >
      Serbo-Croatian
    </option>
    <option
      value="source"
    >
      German
    </option>
    <option
      value="repeatAt"
    >
      Repetition date
    </option>
  </select>
</div>
`;

exports[`should render WordListFilters with changed filters correclty 1`] = `
<div>
  <input
    onChange={[Function]}
    type="text"
    value="t"
  />
  <select
    onChange={[Function]}
    value="source"
  >
    <option
      value="destination"
    >
      Serbo-Croatian
    </option>
    <option
      value="source"
    >
      German
    </option>
    <option
      value="repeatAt"
    >
      Repetition date
    </option>
  </select>
</div>
`;
